src/matrices.jl

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    UnitMatrix{T}(d)
+    UnitColumn{T}(d)
```

Comment:
Let's rename these to `OnesMatrix` and `OnesColumn` to remove ambiguity with the unit element.

Reply:
Actually, let's simply call it `JMatrix` and `JColumn` to match the notation. Otherwise we will have a difficult time finding good names for the other matrices below.